geomstats/geometry/lower_triangular_matrices.py

```diff
@@ -55,8 +61,13 @@ def belongs(self, point, atol=gs.atol):
         """
         belongs = super().belongs(point)
         if gs.any(belongs):
-            is_lower_triangular = Matrices.is_lower_triangular(point, atol)
-            return gs.logical_and(belongs, is_lower_triangular)
+            if self.strictly_lower:
+                is_strictly_lower_triangular = \
+                    Matrices.is_strictly_lower_triangular(point, atol)
+                return gs.logical_and(belongs, is_strictly_lower_triangular)
+            else:
+                is_lower_triangular = Matrices.is_lower_triangular(point, atol)
+                return gs.logical_and(belongs, is_lower_triangular)
         return belongs
 
     @staticmethod
```

Comment:
the fact it requires `self.strictly_lower` requires this to become a normal method and not static anymore.